### PR TITLE
Kdesktop 1245 exclude files folders with name length 255 characters

### PR DESCRIPTION
**Pull request by herve-er**

### CODE REVIEW

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (ClementKunz) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp:124]: You could also filter out EDIT operations

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (ClementKunz) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp:190]: Perhaps you can encapsulate this part in a dedicated method to improve`blacklistNode`readability

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerutility.cpp
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerutility.cpp:65]:
```suggestion
    const SyncName sub = name.stem().native().substr(0, maxNameLengh - suffix.size() - name.extension().native().size());
```

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerutility.cpp
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerutility.cpp:135]:
```suggestion
    return (name.size() > maxNameLengh);
```

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerutility.cpp
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerutility.cpp:134]:
```suggestion
bool PlatformInconsistencyCheckerUtility::isTooLong(const SyncName &name) const {
```

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp:39]: **Suggestion.** 
Define a small helper method `ExitCode checkTree(const ReplicaSide side)` that calls `checkRemoteTree` and `checkLocalTree` depending on its argument.

Comment (luc-guyot-infomaniak) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp:39]:
```suggestion
   const ExitCode exitCodeRemote = checkRemoteTree(_syncPal->updateTree(ReplicaSide::Remote)->rootNode(),
```

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp:42]:
```suggestion
    const ExitCode exitCodeLocal = checkLocalTree(_syncPal->updateTree(ReplicaSide::Local)->rootNode(),
```

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp:45]: **Suggestion.** Make this logging happen within `checkTree(const ReplicaSide side)`. This way you will reduce redundancy.

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp:128]:
```suggestion
   const bool hasNewName = localNode->hasChangeEvent(OperationType::Create) || localNode->hasChangeEvent(OperationType::Move);
    if (hasNewName && PlatformInconsistencyCheckerUtility::instance()->isTooLong(localNode->name())) {
```

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp:135]:
```suggestion
    auto childIt = localNode->children().begin();
```

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp:136]:
```suggestion
    for (; childIt != localNode->children().end(); childIt++) {
```

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp:149]:
```suggestion
       const ExitCode exitCode = checkLocalTree(childeIt->second, parentPath / localNode->name());
```

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp:168]:
```suggestion
                          L"Excluding local item with " << Utility::formatSyncPath(absoluteLocalPath) << L".");
```

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp:170]: Create a small helper function. Naming it will already help understand the expected outcome.

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp:179]:
```suggestion
                          L"Failed to retrieve dbId for local node: " << Utility::formatSyncPath(absoluteLocalPath));
```

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp:182]:
```suggestion
                // Remove the node and its descendants recursively from DB if it the node exists (else, there is nothing to remove).
```

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp:183]:
```suggestion
                LOGW_WARN(_logger, L"Failed to delete node from DB: " << Utility::formatSyncPath(absoluteLocalPath));
```

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp:186]: A small helper would be nice, mostly to document what is the the expected outcome and why.

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp:203]:
```suggestion
                                                << Utility::formatSyncPath(node->getPath()) << L" because "
```

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp:206]: A small helper function (might be a lambda) would avoid code duplication.

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp
Comment (luc-guyot-infomaniak) [inline test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp:309]: We use this constructor so many times with the first argument equal to `std::nullopt` that we could define a new constructor with a shorter list of arguments; at least for the unit tests.

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (herve-er) [reply src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp]: Done here: [dbc6c82](https://github.com/Infomaniak/desktop-kDrive/pull/347/commits/dbc6c82d22f89ad9c489f6abaafc65fd89815435)

File: test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp
Comment (herve-er) [reply test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp]: Done here: [71a448f](https://github.com/Infomaniak/desktop-kDrive/pull/347/commits/71a448f59ceecf81771afb75e80875e7df4af8cd)
Merge state
Review required
At least 1 approving review is required by reviewers with write access. 
[View](https://github.com/Infomaniak/desktop-kDrive/pull/347/files#conversations-menu)
Unresolved conversations
9 conversations must be resolved before merging.
Some checks haven’t completed yet
1 queued and 3 expected checks
@github-actions
Dependent Issues / check (pull_request_target) Queued — Waiting to run this check...
[Details](https://github.com/Infomaniak/desktop-kDrive/actions/runs/11341688325/job/31540470169?pr=347)
Dependent Issues Expected — Waiting for status to be reported
Required
SonarCloud Code Analysis Expected — Waiting for status to be reported
Required
build-kDrive Expected — Waiting for status to be reported
Required
This branch is out-of-date with the base branch
Merge the latest changes from develop into this branch.
 Merge without waiting for requirements to be met (bypass branch protections)

 Automatically merge when all requirements are met. [Learn more](https://docs.github.com/github/collaborating-with-issues-and-pull-requests/automatically-merging-a-pull-request)
@herve-er


Add a comment
Comment
 
Add your comment here...
 
Remember, contributions to this repository should follow our GitHub Community Guidelines.
 ProTip! Add .patch or .diff to the end of URLs for Git’s plaintext views.
Reviewers
@[ClementKunz](https://github.com/ClementKunz)
ClementKunz

@[luc-guyot-infomaniak](https://github.com/luc-guyot-infomaniak)
luc-guyot-infomaniak

At least 1 approving review is required to merge this pull request.

Still in progress?
Assignees
@[herve-er](https://github.com/herve-er)
herve-er

Labels
None yet
Projects
None yet
Milestone
No milestone
Development
Successfully merging this pull request may close these issues.

None yet


Notifications
Customize
You’re receiving notifications because you were assigned.
3 participants
@herve-er
@ClementKunz
@luc-guyot-infomaniak

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (herve-er) [reply src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp]: Done here: [c020720](https://github.com/Infomaniak/desktop-kDrive/pull/347/commits/c020720b986aaa8c951dc2bc2d6af8629d442000)

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (herve-er) [reply src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp]: [c020720](https://github.com/Infomaniak/desktop-kDrive/pull/347/commits/c020720b986aaa8c951dc2bc2d6af8629d442000)

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (herve-er) [reply src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp:169]: Could you please provide more insight about what you are thinking about. I do not see what the small helper functions should do here

File: src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp
Comment (herve-er) [reply src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp]: Done here: [4f28977](https://github.com/Infomaniak/desktop-kDrive/pull/347/commits/4f28977c75d813569308efb36f9e5f22fea5f921)

**Review by ClementKunz — APPROVED** [review]

**Review by luc-guyot-infomaniak — APPROVED** [review]

**Pull request merged**
